scripts/devnet/gen-devnet.sh

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/env bash
+# (c) Cartesi and individual authors (see AUTHORS)
+# SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+set -o nounset
+set -o pipefail
+if [[ "${TRACE-0}" == "1" ]]; then set -o xtrace; fi
+
+script_dir="$( cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly script_dir
+
+. "$script_dir/lib/util.sh"
+. "$script_dir/lib/anvil.sh"
+. "$script_dir/lib/contracts.sh"
+
+################################################################################
+# Configuration
+ROLLUPS_CONTRACTS_VERSION="1.1.0"
+DEVNET_RPC_URL="http://localhost:8545"
+DEVNET_FOUNDRY_ACCOUNT_0_ADDRESS="0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+DEVNET_FOUNDRY_ACCOUNT_0_PRIVATE_KEY="0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+# keccak256("salt")
+DEVNET_DEFAULT_SALT="0xa05e334153147e75f3f416139b5109d1179cb56fef6a4ecb4c4cbc92a7c37b70"
+readonly ROLLUPS_CONTRACTS_VERSION \
+    DEVNET_RPC_URL \
+    DEVNET_FOUNDRY_ACCOUNT_0_ADDRESS \
+    DEVNET_FOUNDRY_ACCOUNT_0_PRIVATE_KEY \
+    DEVNET_DEFAULT_SALT
+
+# Defaults
+devnet_anvil_state_file=$(realpath "./anvil_state.json")
+devnet_deployment_file=$(realpath "./deployment.json")
+template_hash_file=""
+VERBOSE=""
+forge_remappings_file="$script_dir/remappings.txt"
+readonly forge_remappings_file
+
+# Deployment info, which will be gathered during processing
+declare -A deployment_info
+
+################################################################################
+# Utility functions
+################################################################################
+usage()
+{
+   echo "Generate devnet for testing the Cartesi Rollups Node"
+   echo
+   echo "Usage: $0 [options]"
+   echo
+   echo "OPTIONS:"
+   echo
+   echo "    -t template-hash-file"
+   echo "        Mandatory Cartesi Machine template hash file"
+   echo "    -a"
+   echo "        Path for output anvil state file"
+   echo "    -d"
+   echo "        Path for deployment information file"
+   echo "    -v"
+   echo "        Verbose mode"
+   echo "    -h"
+   echo "        Show this message"
+   echo
+
+   exit 0
+}
+
+################################################################################
+# Finish processing and clean up environment
+finish() {
+    verbose "finishing up"
+
+    if [ -n "$work_dir" ]; then
```

Comment:
If `check_error` gets called, this will never be executed, leaving the work dir behind.

Reply:
Not a problem for now because this script will be run inside docker build.

Reply:
Not really. This trap should guarantee that `finish` is executed no matter what:

https://github.com/cartesi/rollups-node/pull/168/files#diff-39ace69d7418eeb86b4569890b7a7f56e6a334ee8b235d854c5dc3d4cae7389cR323